core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/SyncMongoClientFactory.java

```diff
@@ -33,7 +33,7 @@ public static MongoClient getClient(
       throw new IllegalArgumentException("StitchAppClient not configured with a data directory");
     }
 
-    final String userId = appInfo.getAuthMonitor().isLoggedIn()
+    final String userId = appInfo.getAuthMonitor().tryIsLoggedIn()
```

Comment:
[q] does this need to be a `tryIsLoggedIn`? I'm concerned that this would cause a real user to end up having their data be in an `"unbound"` data directory

Reply:
In the current state of our codebase, it would not be possible for this to happen. However, maybe I'll make a note here for "in the future".